src/algorithms/MyArrayList.java

```diff
@@ -0,0 +1,151 @@
+package src.algorithms;
+
+import java.io.Serializable;
+import java.util.*;
+
+public class MyArrayList<T>  implements Iterable<T>, Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private static final int DEFAULT_CAPACITY = 10;
+    transient Object[] elements;
+    private int size;
+
+    public MyArrayList() {
+        this.elements = new Object[DEFAULT_CAPACITY];
+    }
+
+    public MyArrayList(int initialCapacity) {
+        if (initialCapacity >= 0) {
+            this.elements = new Object[initialCapacity];
+        } else {
+            throw new IllegalStateException("Начальная емкость (initialCapacity) не может быть меньше или равен нулю");
+        }
+    }
+    public int size() {
+        return size;
+    }
+
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    private Object[] capacityGrowth(int minCapacity ) {
+        int capacity = elements.length;
+        if (capacity > 0) {
+            Object[] newCapacity = new Object[(elements.length * 2)];
+            System.arraycopy(elements, 0, newCapacity, 0, elements.length);
+            return newCapacity;
+        }else{
+            return elements = new Object[Math.max(DEFAULT_CAPACITY, minCapacity)];
+        }}
+
+    private Object[] capacityGrowth() {
+        return capacityGrowth(size + 1);
+    }
+
+    public boolean add(T object){
+        if (size == elements.length){
```

Comment:
Еще один пробел потерян
```suggestion
        if (size == elements.length) {
```